### PR TITLE
[Merged by Bors] - feat: `WithAbs` type synonym

### PR DESCRIPTION
**Pull request by smmercuri**
This PR contains the type synonym `WithAbs`. This is a type synonym for a semiring `R` that depends on an absolute value `v : AbsoluteValue R S`, where `S` is an `OrderedSemiring`. This is used in order to handle ambiguities arising from multiple sources of instances. For example, each infinite place on a number field defines distinct uniform structures via their associated real absolute values. The `WithAbs` type synonym allows the type class inferencing system to automatically infer such dependent instances, making it simpler to define the Archimedean completions of a number field. See #16483 for this application along with prior feedback. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by mattrobball — CHANGES_REQUESTED** [review]

File: Mathlib/Topology/UniformSpace/AbsoluteValue.lean [hunk 1]
```diff
@@ -5,6 +5,7 @@ Authors: Patrick Massot
 -/
 import Mathlib.Algebra.Order.AbsoluteValue
 import Mathlib.Algebra.Order.Field.Basic
+import Mathlib.Analysis.Normed.Field.Basic
```
[the remark this answers is not in the document]
Comment (smmercuri) [reply Mathlib/Topology/UniformSpace/AbsoluteValue.lean]: Thanks, have now moved those to `Mathlib.Analysis.Normed.Field.Basic`.

**Review by mattrobball — APPROVED** [review]
I think this approach makes sense. Since there is some discussion on Zulip, I will just approve and not merge at the moment.

**Comment by riccardobrasca** [comment]
Can you please fix the error?

**Comment by riccardobrasca** [comment]
It seems we agree this approach makes sense, let's merge it and see if something needs to be improved. Thanks!

bors mergr

**Comment by riccardobrasca** [comment]
ops.

bors merge

**Comment by faenuccio** [comment]
I know this is now closed, but as in my comments for #16483 I would suggest to rename `WithAbs` to `WithAbsRing`.

**Pull request closed without merging**